src/plugin_deno_resolver.ts

```diff
@@ -118,16 +123,25 @@ export function denoResolverPlugin(
         // We can then resolve the specifier relative to the referrer URL, using
         // the workspace resolver.
         const resolved = new URL(
-          resolver!.resolve(args.path, referrer.href),
+          resolver!.resolve(originalPath, referrer.href),
         );
 
         // Now pass the resolved specifier back into the resolver, for a second
         // pass. Now plugins can perform any resolution they want on the fully
         // resolved specifier.
         const { path, namespace } = urlToEsbuildResolution(resolved);
+        if (namespace === "data" && !DATA_FILTER.test(path)) {
```

Comment:
The resolver plugin doesn't have a filter for onResolve. This ensures it doesn't try to resolve data urls it cannot handle, like for an svg.